pyproject.toml

```diff
@@ -73,27 +73,28 @@ select = [
 ignore = [
     "D104", # Missing docstring in public package
     "D100", # Missing docstring in public module
-    "D211", # No blank line before class
     "D213", # Multiline summary second line
     "PD901", # Avoid using 'df' for pandas dataframes. Perfectly fine in functions with limited scope
-    "ANN201", # Missing return type annotation for public function (makes no sense for NoneType return types...)
-    "ANN101", # Missing type annotation for `self`
     "ANN204", # Missing return type annotation for special method
     "ANN002", # Missing type annotation for `*args`
     "ANN003", # Missing type annotation for `**kwargs`
     "D105", # Missing docstring in magic method
     "D203", # 1 blank line before after class docstring
-    "D204", # 1 blank line required after class docstring
     "D413", # 1 black line after parameters
     "SIM108", # Simplify if/else to one line; not always clearer
-    "D206", # Docstrings should be indented with spaces; unnecessary when running ruff-format
-    "E501", # Line length too long; unnecessary when running ruff-format
-    "W191", # Indentation contains tabs; unnecessary when running ruff-format
```

Comment:
We're still using `ruff format` here, should these be kept? See https://docs.astral.sh/ruff/rules/tab-indentation/#formatter-compatibility

Reply:
I just swapped the running order and run ruff-format first, so these lines become inconsequential